roq-clientlib/src/main/java/org/roqmessaging/clientlib/factory/IRoQLogicalQueueFactory.java

```diff
@@ -27,10 +27,11 @@
 	 * Creates a logical queue.
 	 * @param queueName the queue name
 	 * @param targetAddress the address to create the queue
+	 * @param recoveryMod the q must be recovered
 	 * @throws IllegalStateException if a queue already exist with this name, the name must be unique for 
 	 * the complete cluster.
 	 */
-	public boolean createQueue(String queueName, String targetAddress) throws IllegalStateException;
+	public int createQueue(String queueName, String targetAddress, boolean recoveryMod) throws IllegalStateException;
```

Comment:
Is there no better way to return an error code than using hard-coded integers? enums? custom error classes?